src/System.Net.Security/src/System/Net/ContextFlagsAdapterPal.Unix.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace System.Net
+{
+    internal static class ContextFlagsAdapterPal
+    {
+        private struct ContextFlagMapping
+        {
+            public readonly Interop.NetSecurityNative.GssFlags GssFlags;
+            public readonly ContextFlagsPal ContextFlag;
+
+            public ContextFlagMapping(Interop.NetSecurityNative.GssFlags gssFlag, ContextFlagsPal contextFlag)
+            {
+                GssFlags = gssFlag;
+                ContextFlag = contextFlag;
+            }
+        }
+
+        private static readonly ContextFlagMapping[] s_contextFlagMapping = new[]
+        {
+            // GSS_C_INTEG_FLAG is set if either AcceptIntegrity (used by server) or InitIntegrity (used by client) is set
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_INTEG_FLAG, ContextFlagsPal.AcceptIntegrity | ContextFlagsPal.InitIntegrity),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_CONF_FLAG, ContextFlagsPal.Confidentiality),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_IDENTIFY_FLAG, ContextFlagsPal.InitIdentify),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_MUTUAL_FLAG, ContextFlagsPal.MutualAuth),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_REPLAY_FLAG, ContextFlagsPal.ReplayDetect),
+            new ContextFlagMapping(Interop.NetSecurityNative.GssFlags.GSS_C_SEQUENCE_FLAG, ContextFlagsPal.SequenceDetect)
+        };
+
+
+        internal static ContextFlagsPal GetContextFlagsPalFromInterop(Interop.NetSecurityNative.GssFlags gssFlags)
+        {
+            ContextFlagsPal flags = ContextFlagsPal.Zero;
+            foreach (ContextFlagMapping mapping in s_contextFlagMapping)
+            {
+                if ((gssFlags & mapping.GssFlags) != 0)
```

Comment:
Should this be ```!= 0``` or ```== mapping.GssFlags```?  I'm wondering if it's possible some of the flags may have multiple bits set, in which case it's possible for the ```&``` to result in a non-zero value that doesn't equal mapping.GssFlags.